brainreg_napari/register.py

```diff
@@ -409,14 +448,77 @@ def run(
+        into the requested atlas orientation, compute the average projection and displays it alongside the atlas. It
+        is then super easy for the super to identify which dimension should be swapped and avoid running the pipeline
+        on wrongly aligned data.
+        """
+
+        # Get viewer object
+        viewer = getattr(widget, "viewer").value
+        brain_geometry = getattr(widget, "brain_geometry").value
+
+        # Remove previous average projection layer if needed
+        ind_pop = []
+        for i, layer in enumerate(viewer.layers):
+            if layer.name in ['Ref. proj. 0', 'Ref. proj. 1', 'Ref. proj. 2',
+                              'Input proj. 0', 'Input proj. 1', 'Input proj. 2']:
+                ind_pop.append(i)
+            else:
+                layer.visible = False
+        for index in ind_pop[::-1]:
+            del viewer.layers[index]
+
+        # Load atlas and gather data
+        atlas = BrainGlobeAtlas('allen_mouse_25um')
+        if brain_geometry.value == "hemisphere_l":
+            atlas.reference[atlas.hemispheres == atlas.left_hemisphere_value] = 0
+        elif brain_geometry.value == "hemisphere_r":
+            atlas.reference[atlas.hemispheres == atlas.right_hemisphere_value] = 0
+        input_orientation = getattr(widget, "data_orientation").value
+        data = getattr(widget, "img_layer").value.data
+        # Transform data to atlas orientation from user input
+        data_remapped = bg.map_stack_to(input_orientation, atlas.orientation, data)
+
+        # Compute average projection of atlas and remapped data
+        u_proj = []
+        u_proja = []
+        s = []
+        for i in range(3):
+            u_proj.append(np.mean(data_remapped, axis=i))
+            u_proja.append(np.mean(atlas.reference, axis=i))
+            s.append(u_proja[-1].shape[0])
+        s = np.max(s)
+
+        # Display all projections with somewhat consistent scaling
+        viewer.add_image(u_proja[0], name='Ref. proj. 0')
+        viewer.add_image(u_proja[1], translate=[0, u_proja[0].shape[1]], name='Ref. proj. 1')
+        viewer.add_image(u_proja[2], translate=[0, u_proja[0].shape[1] + u_proja[1].shape[1]], name='Ref. proj. 2')
+
+        s1 = u_proja[0].shape[0] / u_proj[0].shape[0]
+        s2 = u_proja[0].shape[1] / u_proj[0].shape[1]
+        viewer.add_image(u_proj[0], translate=[s, 0], name='Input proj. 0', scale=[s1, s2])
+        s1 = u_proja[1].shape[0] / u_proj[1].shape[0]
+        s2 = u_proja[1].shape[1] / u_proj[1].shape[1]
+        viewer.add_image(u_proj[1], translate=[s, u_proja[0].shape[1]], name='Input proj. 1', scale=[s1, s2])
+        s1 = u_proja[2].shape[0] / u_proj[2].shape[0]
+        s2 = u_proja[2].shape[1] / u_proj[2].shape[1]
+        viewer.add_image(u_proj[2], translate=[s, u_proja[0].shape[1] + u_proja[1].shape[1]], name='Input proj. 2',
+                         scale=[s1, s2])
```

Comment:
For me, I would prefer different variable names besides `s1`/`s2` after the first time. Looks like this could be done in a more succinct way IMO

Reply:
It was done initially to resize correctly given the voxel size entered by the user. There was however a bug in keeping track of the transformations so I removed it for now. I am planning on working on it once I have time. Right now the scaling is somewhat consistent and allow to check the orientation but edge effect could happen if the input data shape is very different from the reference shape.

I would still advice not sacrificing readability for succinctness.

Reply:
Yeah other than extracting some of this code into a function, I'm pretty happy about how readable it is. Simplicity is v. important for a project that we hope life scientists will contribute to.